src/uucore/src/lib/parser/parse_size.rs

```diff
@@ -55,7 +55,7 @@ pub fn parse_size(size: &str) -> Result<u64, ParseSizeError> {
     // empty string, in which case, the factor is 1.
     let unit = &size[numeric_string.len()..];
     let (base, exponent): (u128, u32) = match unit {
-        "" => (1, 0),
+        "B" | "" => (1, 0),
```

Comment:
Looks like this change is causing a failure in `truncate`, which seems to disallow this:

```
failures:

---- test_truncate::test_invalid_numbers stdout ----
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0X file
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0XB file
current_directory_resolved: 
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils truncate -s 0B file
thread 'test_truncate::test_invalid_numbers' panicked at 'Command was expected to fail.
stdout = 
 stderr = ', tests/common/util.rs:185:9
```